bin/src/command/mod.rs

```diff
@@ -71,34 +68,12 @@ enum CommandMessage {
         worker_id: u32,
     },
     Advancement {
-        request_identifier: RequestIdentifier,
+        client_id: String,
         response: Response,
     },
     MasterStop,
 }
 
-/// identifies a request only within the command server
-/// the request part does NOT get sent to a worker
-#[derive(PartialEq, Eq, Clone, Debug)]
-pub struct RequestIdentifier {
-    /// the client who sent the request (ex: "CL-0")
-    client: String,
-    // the request id (ex: "ID-MAN9QF")
-    request: String,
-}
-
-impl RequestIdentifier {
-    pub fn new<T>(client: T, request: T) -> Self
-    where
-        T: ToString,
```

Comment:
It seems to me that it is better to add a criteria like this `Into<String>` and then call `.into` on object.
Besides, you should use one generic per parameter which allow to put different types as arguments.
```rust
pub fn new<C, R>(client: C, request: R) -> Self
where
        C: Into<String>,
        R: Into<String>,
```

Reply:
Yes, thanks for the reminder!